File: backend/app/schemas/vehicle.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class VehicleCreate(BaseModel):
    company_id: UUID | None = None
    vehicle_no: str
    owner_name: str | None = None
    driver_name: str | None = None
    phone: str | None = None
    vehicle_type: str | None = None
    is_active: bool = True


class VehicleUpdate(BaseModel):
    company_id: UUID | None = None
    vehicle_no: str | None = None
    owner_name: str | None = None
    driver_name: str | None = None
    phone: str | None = None
    vehicle_type: str | None = None
    is_active: bool | None = None


class VehicleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    company_id: UUID | None
    vehicle_no: str
    owner_name: str | None
    driver_name: str | None
    phone: str | None
    vehicle_type: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime

File: backend-spring/src/main/java/com/nearkart/entity/Vehicle.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "vehicles")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "company_id")
    private UUID companyId;

    @Column(name = "vehicle_no", nullable = false)
    private String vehicleNo;

    @Column(name = "owner_name")
    private String ownerName;

    @Column(name = "driver_name")
    private String driverName;

    private String phone;

    @Column(name = "vehicle_type")
    private String vehicleType;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() { OffsetDateTime now = OffsetDateTime.now(); createdAt = now; updatedAt = now; }

    @PreUpdate
    protected void onUpdate() { updatedAt = OffsetDateTime.now(); }
}
